### DISCUSSION TITLE
How can I get the position of the cursor counting from the beginning of the line?

### DISCUSSION THREAD

**Discussion by YumaHJET** [microsoft/vscode-discussions#343] [Extension Development QnA] [answered]
How can I get the position of the cursor counting from the beginning of the line?

**Comment by alefragnani** [comment, marked as answer]
Hi @YumaHJET ,

You could use `vscode.window.activeTextEditor.selection.active.character`, if you intend to take care only the _primary_ selection where the cursor is. 

Otherwise, if you need _multi-selection / multi-cursor_ support, you should look for `vscode.window.activeTextEditor.selections` and loop through the array. Each item in the array will provide you an individual selection/cursor on the multi-cursor scenario you have.

Just for documentation purposes, this is what the VS Code API describes for `Selection` and `Active`:

```ts
    /**
     * Represents a text selection in an editor.
     */
    export class Selection extends Range {
        ...
        /**
         * The position of the cursor.
         * This position might be before or after {@link Selection.anchor anchor}.
         */
        active: Position;
```

Hope this helps

**Reply by YumaHJET** [reply]
Thankyou!
